### PR TITLE
Add tests to check that SLX file versions do not exceed R2020b

### PR DESCRIPTION
**Pull request by H0R5E**
This PR adds some new tests to check whether the versions of any SLX files stored in the repository exceed R2020b.

Having these tests will make sure that incompatible versions are not merged accidentally.

### CODE REVIEW

**Comment by H0R5E** [comment]
The test are revealing 5 files that exceed the maximum version:

- `examples/OSWEC/OSWEC.slx`
- `examples/RM3/RM3.slx`
- `examples/RM3FromSimulink/RM3FromSimulink.slx`
- `source/lib/WEC-Sim/WECSim_Lib_Body_Elements.slx`
- `source/lib/WEC-Sim/WECSim_Lib_Frames.slx model`

```
================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_examples_OSWEC_OSWEC_slx).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/examples/OSWEC/OSWEC.slx model version R2021a exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
  
      Actual Value:
        logical
  
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
  ================================================================================
  .
  ================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_examples_RM3_RM3_slx).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/examples/RM3/RM3.slx model version R2021a exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
  
      Actual Value:
        logical
  
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
  ================================================================================
  .
  ================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_examples_RM3FromSimulink_RM).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/examples/RM3FromSimulink/RM3FromSimulink.slx model version R2021a exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
  
      Actual Value:
        logical
  
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
  ================================================================================
  ...
  ================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_source_lib_WEC_Sim_WECSim_1).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/source/lib/WEC-Sim/WECSim_Lib_Body_Elements.slx model version R2021b exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
  
      Actual Value:
        logical
  
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
  ================================================================================
  ...
  ================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_source_lib_WEC_Sim_WECSim_4).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/source/lib/WEC-Sim/WECSim_Lib_Frames.slx model version R2021b exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
  
      Actual Value:
        logical
  
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
  ================================================================================
```

**Comment by H0R5E** [comment]
I added #920 to track converting the SLX files with the wrong versions.

**Comment by H0R5E** [comment]
@kmruehl, do you think it's worth adding something to the [WEC-Sim Library](https://wec-sim.github.io/WEC-Sim/master/developer/library.html) page about how to save the files to the correct version?

**Comment by kmruehl** [comment]
@H0R5E we require the [WEC-Sim Library to be saved in 2020b](http://wec-sim.github.io/WEC-Sim/master/user/getting_started.html#matlab-requirements), but we've never required model files to also be saved in a prior version of MATLAB. There have been (rare) instances when a model is saved in a newer version of MATLAB (e.g. 2021b), and I'm unable to open it in an older version (e.g. 2020a). However, this is not a common issue, and I do not want to require all WEC-Sim models to be saved in prior versions of MATLAB. I'd prefer to just update the cases causing issues. 

In this instance, it looks like two of the library files were not properly saved in 2020b. I'll update those and see if it resolves the issue. 
- ``source/lib/WEC-Sim/WECSim_Lib_Body_Elements.slx``
- ``source/lib/WEC-Sim/WECSim_Lib_Frames.slx``

**Comment by kmruehl** [comment]
@H0R5E, @jtgrasb is going to make the revision to the WEC-Sim library blocks on dev (via a direct bug-fix push). I don't want to require the WEC-Sim model files to be saved to 2020b, but this check would be helpful to run on the library files, on the `WEC-Sim\source\lib` directory. Can you update this PR so that it only checks the library files?

Thanks!

**Comment by H0R5E** [comment]
> @H0R5E we require the [WEC-Sim Library to be saved in 2020b](http://wec-sim.github.io/WEC-Sim/master/user/getting_started.html#matlab-requirements), but we've never required model files to also be saved in a prior version of MATLAB.

Sorry, I'm going to have to disagree with this approach. If we are supporting users running R2020b, then they should be able to open the example files. For example, when I try to open `\examples\RM3\RM3.slx`, I get this error:

wec-sim_slx_version_error

It's very low effort to save the files in the correct format, so I don't understand why it's not worth doing to ensure compatibility for all users.

At a minimum, we should put a warning in the docs saying that the example files can only be opened with a specific version.

**Comment by H0R5E** [comment]
So, another option is to get our users to turn off the compatibility check when opening Simulink files saved in newer versions. `RM3.slx` still throws this warning after being opened:

```
=== Model Load (Elapsed: 0.176 sec) ===
    Warning:Model 'RM3' was created with a newer version (R2021a) of Simulink
    To create a model that is compatible with this version of Simulink, load the model in Simulink R2021a and select Save > Export Model to > Previous Version.
    Warning:Warning: Cannot load version 'R2021a_202009111111' of family 'sltp_mm'.
```

In this case, we should document the process of setting the option in the Simulink preferences.

**Comment by kmruehl** [comment]
@H0R5E do you know is there is a way to turn off that option programmatically? The WEC-Sim team discussed this issue. We only way to require the library to be saved in a prior version of MATLAB, but would like users to be able to open model files saved in a newer version of MATLAB.

**Comment by H0R5E** [comment]
> @H0R5E do you know is there is a way to turn off that option programmatically?

Yes, the command is `set_param(0,'ErrorIfLoadNewModel','off')`

**Comment by kmruehl** [comment]
`set_param(0,'ErrorIfLoadNewModel','off')` is a global setting. It only needs to be run once. 

There are 3 options of how to implement this (not listed in order of preference):
1. add a note in the documentation with instructions on how to change the global setting (this would be run optionally)
2. add it to wec-sim install (e.g. `startup.m`) with a comment about it being a global setting (this would be run once)
3. add it to simulation class (whether other simulink settings are changed), and add a warning noting the change to a global setting (this would be run multiple times unless there's an if/else statement)

**Comment by H0R5E** [comment]
So, the command above is called [here](https://github.com/WEC-Sim/WEC-Sim/blob/f9ac900f3336548ee4ca40af73d98d9d27564fa3/source/functions/initializeWecSim.m#L461) in `initializeWecSim.m`.

I really don't think this is the best place for it, as you need to run WEC-Sim before you can open the model.

🛒 before 🐴.

**Comment by H0R5E** [comment]
This command has been inside the `wecSim` script for a very long time. It was added by @yuyihsiang on [2017/08/16](https://github.com/WEC-Sim/WEC-Sim/commit/bddbfbbae9ccf38bb4fdf7c6ee56b3851a643277).

The problem is that if we removed this command and it was not run before calling WEC-Sim then WEC-Sim would fail if we were trying to run a newer model. Therefore, you could argue it's a *necessary* part of the installation process.

So I think putting it into `startup.m` makes the most sense to me. You can then open newer model files without running WEC-Sim first and WEC-Sim won't break when running newer model files if you haven't set the preference manually.

**Comment by H0R5E** [comment]
So, I've gone ahead and moved `set_param(0,'ErrorIfLoadNewModel','off')` from `initializeWecSim.m` to `AddWecSimSource.m` and reduced the Simulink file version testing to the `source` folder.

Happy to change again if not what you want. I just thought it would be useful to see what it looked like in this configuration.

**Comment by kmruehl** [comment]
@H0R5E do you know why these tests are failing?

**Comment by H0R5E** [comment]
Hi, the test failure being generated is:

```
 ================================================================================
  Verification failed in SLXVersionTest/testSLXVersion(Paths=p__home_runner_work_WEC_Sim_WEC_Sim_source_lib_WEC_Sim_WECSim_2).
      ----------------
      Test Diagnostic:
      ----------------
      /home/runner/work/WEC-Sim/WEC-Sim/source/lib/WEC-Sim/WECSim_Lib_Cables.slx model version R2021b exceeds R2020b
      ---------------------
      Framework Diagnostic:
      ---------------------
      verifyTrue failed.
      --> The value must evaluate to "true".
      
      Actual Value:
        logical
      
         0
      ------------------
      Stack Information:
      ------------------
      In /home/runner/work/WEC-Sim/WEC-Sim/tests/devTests/SLXVersionTest.m (SLXVersionTest.testSLXVersion) at 16
```

The most important part being `source/lib/WEC-Sim/WECSim_Lib_Cables.slx model version R2021b exceeds R2020b`, indicating that `WECSim_Lib_Cables.slx` is not saved in a R2020b version.

**Comment by kmruehl** [comment]
@dforbush2 can you save the cable library file to 2020b?

File: source/functions/initializeWecSim.m [hunk 1]
```diff
@@ -463,7 +463,7 @@
 warning('off','MATLAB:printf:BadEscapeSequenceInFormat');
 warning('off','Simulink:blocks:DivideByZero');
 warning('off','sm:sli:setup:compile:SteadyStateStartNotSupported')
-set_param(0, 'ErrorIfLoadNewModel', 'off')
+
 % Load parameters to Simulink model
```
Comment (kmruehl) [inline source/functions/initializeWecSim.m:468]: My only concern about removing it here, is that not everyone uses the `addWecSimSource` script each time they open MATLAB. I think we should keep it in both places. I'm going to add it back here again too.

File: addWecSimSource.m [hunk 2]
```diff
@@ -4,4 +4,7 @@
 wecSimSource = fullfile(pwd,'source');
 addpath(genpath(wecSimSource));
 
-clear wecSimSource
\ No newline at end of file
+% Allow opening of Simulink models saved in a newer version
+set_param(0, 'ErrorIfLoadNewModel', 'off')
+
```
Comment (kmruehl) [inline addWecSimSource.m:9]: This solution only works if users execute this script, which they may not do. For example, I have the source always added and never run this script. I think we should keep it in both places, and will make this revision.

**Review by kmruehl — APPROVED** [review]
@H0R5E I'm going to make this minor change and then merge the PR once the tests pass

**Comment by kmruehl** [comment]
@H0R5E you can delete your branch if you like

**Pull request merged**